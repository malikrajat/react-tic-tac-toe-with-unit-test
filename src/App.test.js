import React from 'react';
import ReactDOM from 'react-dom';
import { shallow, configure, mount } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import renderer from 'react-test-renderer';

import App from './App';

configure({ adapter: new Adapter() });

describe("App component test ", () => {
  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<App />, div);
    ReactDOM.unmountComponentAtNode(div);
  });

  it("should render App component", () => {
    const wrapper = shallow(<App />);
  });

  it("Find class name by shallow mrthod", () => {
    const wrapper = shallow(<App />);
    expect(wrapper.hasClass('App')).toEqual(true);
  });

  it('snapshot renders', () => {
    const component = renderer.create(<App/>);
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});