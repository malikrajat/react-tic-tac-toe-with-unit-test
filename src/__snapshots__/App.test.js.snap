// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App component test  snapshot renders 1`] = `
<div
  className="App"
>
  <div>
    Winner is : 
    
  </div>
  <div
    className="row"
  >
    <span
      className="sell "
      onClick={[Function]}
    >
      
    </span>
    <span
      className="sell "
      onClick={[Function]}
    >
      
    </span>
    <span
      className="sell "
      onClick={[Function]}
    >
      
    </span>
  </div>
  <div
    className="row"
  >
    <span
      className="sell "
      onClick={[Function]}
    >
      
    </span>
    <span
      className="sell "
      onClick={[Function]}
    >
      
    </span>
    <span
      className="sell "
      onClick={[Function]}
    >
      
    </span>
  </div>
  <div
    className="row"
  >
    <span
      className="sell "
      onClick={[Function]}
    >
      
    </span>
    <span
      className="sell "
      onClick={[Function]}
    >
      
    </span>
    <span
      className="sell "
      onClick={[Function]}
    >
      
    </span>
  </div>
</div>
`;
